src/main/java/org/json/simple/parser/JSONParser.java

```diff
@@ -121,7 +121,7 @@ public Object parse(Reader in, ContainerFactory containerFactory) throws IOExcep
 					switch(token.type){
 					case Yytoken.TYPE_VALUE:
 						status=S_IN_FINISHED_VALUE;
-						statusStack.addFirst(new Integer(status));
+						statusStack.addFirst(Integer.valueOf(status));
```

Comment:
Your status list should be parametrized to store Integer values then you do not need Integer.valueOf either